cpp/log/cert.cc

```diff
@@ -887,10 +963,23 @@ bool ValidateRedactionSubjectAltNames(STACK_OF(GENERAL_NAME) *
 }  // namespace
 
 
+util::Status Cert::SubjectAltNames(vector<string>* dns_alt_names) const {
+  ScopedGENERAL_NAMEStack subject_alt_names(
+      static_cast<STACK_OF(GENERAL_NAME)*>(X509_get_ext_d2i(
+          x509_.get(), NID_subject_alt_name, nullptr, nullptr)));
+
+  if (subject_alt_names) {
```

Comment:
The helper was extracted from ValidateRedactionSubjectAltNames() originally.  I intended to then go back and refactor, but forgot.

You're right, there's a difference: the original ValidateRedactionSubjectAltNames() did extra stuff at the back end of its 'for' loop.

I figured that it would be acceptable to do those extra things as a second phase for-loop in order to promote the re-use of the extracted method.  The behaviour in the error case will be a little different (the vector<string> will have more content in) but I thought that probably wouldn't matter.